### PR TITLE
Test SimpleNotificationDisplayer implementation

### PR DESCRIPTION
**Pull request by hectorespert**
Add test for SimpleNotificationDisplayer implementation and resolve some warnings.

### CODE REVIEW

File: platform/openide.awt/test/unit/src/org/openide/awt/NotificationDisplayerTest.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+import org.junit.Test;
+import org.netbeans.junit.NbTestCase;
+import org.openide.awt.NotificationDisplayer.Category;
+import org.openide.awt.NotificationDisplayer.Priority;
+
+/**
+ *
+ * @author Hector Espert
+ */
+public class NotificationDisplayerTest extends NbTestCase {
```
Comment (eirikbakke) [inline platform/openide.awt/test/unit/src/org/openide/awt/NotificationDisplayerTest.java]: Maybe rename to SimpleNotificationDisplayerTest.

File: platform/openide.awt/test/unit/src/org/openide/awt/NotificationDisplayerTest.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+    public NotificationDisplayerTest(String name) {
+        super(name);
+    }
+    
+    @Override
+    public void setUp() {
+        notificationDisplayer = NotificationDisplayer.getDefault();
+        assertNotNull(notificationDisplayer);
+        assertEquals("SimpleNotificationDisplayer", notificationDisplayer.getClass().getSimpleName());
```
Comment (eirikbakke) [inline platform/openide.awt/test/unit/src/org/openide/awt/NotificationDisplayerTest.java]: Should we make assumptions about which NotificationDisplayer is the default? That's not specified by the API.

Are these tests that should pass for _any_ implementation of NotificationDisplayer?

File: platform/openide.awt/test/unit/src/org/openide/awt/NotificationDisplayerTest.java [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    }
+
+    @Test
+    public void testNotify_4args() {
+        Notification notification = notificationDisplayer.notify("title", null, "details", null);
+        assertNotNull(notification);
+        assertEquals("NotificationImpl", notification.getClass().getSimpleName());
```
Comment (eirikbakke) [inline platform/openide.awt/test/unit/src/org/openide/awt/NotificationDisplayerTest.java]: That's an implementation detail--ideally the tests should not break if you change the implementation while following the API specs.

File: platform/openide.awt/test/unit/src/org/openide/awt/NotificationDisplayerTest.java [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        assertEquals("title", StatusDisplayer.getDefault().getStatusText());
+    }
+
+    @Test
+    public void testNotify_6args_4() {
+        Notification notification = notificationDisplayer.notify("title", null, (JComponent)null, (JComponent)null, Priority.LOW, "default_category_error");
+        assertNotNull(notification);
+        assertEquals("NotificationImpl", notification.getClass().getSimpleName());
+        assertEquals("title", StatusDisplayer.getDefault().getStatusText());
```
Comment (eirikbakke) [inline platform/openide.awt/test/unit/src/org/openide/awt/NotificationDisplayerTest.java]: These tests are repetitive--maybe factor out into a single testNotify(String expectedTitle, Notification notification) method.

File: platform/openide.awt/test/unit/src/org/openide/awt/NotificationDisplayerTest.java (hunk 2 above)
Comment (blackleg) [reply platform/openide.awt/test/unit/src/org/openide/awt/NotificationDisplayerTest.java]: > Should we make assumptions about which NotificationDisplayer is the default? That's not specified by the API.

It doesn't assume that `SimpleNotificationDisplayer` is the default NotificationDisplayer, only is for testing that if any other is not present, fallback implementation is loaded.

**Pull request merged**